File: src/hackerrank/30 Days of Code/Day 06 — Let's Review/solution.js
// Source: https://www.hackerrank.com/challenges/30-review-loop/problem
// Author: Frank Faustino
// Date  : 2018-01-21
// Lang  : JavaScript

const processData = (input) => input
  .split('\n')
  .slice(1)
  .map(v => {
    let even = ''
    let odd = ''
    v.split('').filter((v, i) => (i % 2 === 0 ? (even += v) : (odd += v)))
    console.log(even + ' ' + odd)
  })

process.stdin.resume()
process.stdin.setEncoding('ascii')
_input = ''
process.stdin.on('data', function (input) {
  _input += input
})

process.stdin.on('end', function () {
  processData(_input)
})

File: src/hackerrank/30 Days of Code/Day 06 — Let's Review/solution.py
# Source: https://www.hackerrank.com/challenges/30-review-loop/problem
# Author: Frank Faustino
# Date:   2018-01-21
# Lang:   Python 3

for i in range(int(input())):
    s = input()
    print(s[::2], s[1::2])

# s[::2]  every 2nd element starting from 0 to length -1
# s[1::2] every 2nd element starting from 1 to length -1